### PR TITLE
Support DS in service-archetype module [ECR-3595]

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview
Support DS in service-archetype module.

---
See: https://jira.bf.local/browse/ECR-3595

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/service-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -123,12 +124,38 @@
...
+        <artifactId>maven-failsafe-plugin</artifactId>
+        <version>2.22.2</version>
+        <configuration>
+          <includes>
+            <include>**/*IntegrationTest.java</include>
+          </includes>
+          <argLine>-Djava.library.path=${nativeLibPath}</argLine>
+          <systemPropertyVariables>
+            <it.artifactFilename>${artifactFilename}.jar</it.artifactFilename>
+            <it.artifactId>${groupId}:${artifactId}:${version}</it.artifactId>
```
Comment (dmitry-timofeev) [inline exonum-java-binding/service-archetype/src/main/resources/archetype-resources/pom.xml]: Those properties are deprecated and lead to warnings.

Comment (dmitry-timofeev) [inline exonum-java-binding/service-archetype/src/main/resources/archetype-resources/pom.xml]: Please extract this and https://github.com/exonum/exonum-java-binding/pull/1229/files#diff-77cf287dc69b3029b3d73154618ed696R183 
in a property.

File: exonum-java-binding/service-archetype/src/main/resources/archetype-resources/pom.xml [hunk 2]
```diff
@@ -20,6 +20,7 @@
     <hamcrest.version>2.1</hamcrest.version>
     <mockito.version>3.1.0</mockito.version>
     <exonum.version>0.9.0-SNAPSHOT</exonum.version>
+    <artifactFilename>${artifactId}-${version}-artifact</artifactFilename>
```
Comment (dmitry-timofeev) [inline exonum-java-binding/service-archetype/src/main/resources/archetype-resources/pom.xml]: See below.

**Comment by dmitry-timofeev** [comment]
Please also make the CI build and test the archetype module.

File: exonum-java-binding/service-archetype/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (MakarovS) [reply exonum-java-binding/service-archetype/src/main/resources/archetype-resources/pom.xml]: Fixed it in `cryptocurrency-demo` as well

**Pull request merged**